src/prices/KyberPriceFeed.sol

```diff
@@ -2,337 +2,368 @@ pragma solidity 0.6.1;
         }
         lastUpdate = block.timestamp;
-        emit PriceUpdate(assets, newPrices);
+        emit PriceUpdate(registeredAssets, newPrices);
     }
 
-    function setUpdaterToRegistryOwner() external {
-        require(msg.sender == UPDATER, "Only current updater can do this");
-        UPDATER = REGISTRY.owner();
+    /// @dev Registry owner can set arbitrary prices here
+    function backupUpdate(
+        address[] calldata _assets,
+        uint256[] calldata _prices
+    )
+        external
+        onlyRegistryOwner
+    {
+        require(
+            _assets.length == _prices.length,
+            "backupUpdate: Passed array lengths unequal"
+        );
+        address[] memory registeredAssets = registry.getRegisteredAssets();
+        for (uint256 i; i < registeredAssets.length; i++) {
+            require(
+                _assets[i] == registeredAssets[i],
+                "backupUpdate: Passed asset does not match registered asset"
+            );
+            if (registeredAssets[i] == registry.nativeAsset()) {
+                require(
+                    _prices[i] == 1 ether,
+                    "backupUpdate: Native asset must have price of 1 ether"
+                );
+            }
+            prices[registeredAssets[i]] = _prices[i];
+        }
+        lastUpdate = block.timestamp;
+        emit BackupPriceUpdate(registeredAssets, _prices);
+    }
+
+    function setUpdater(address _updater) external onlyRegistryOwner {
+        updater = _updater;
+        emit UpdaterSet(_updater);
     }
 
-    function setUpdater(address _updater) external {
-        require(msg.sender == REGISTRY.owner(), "Only registry owner can set");
-        UPDATER = _updater;
+    function setRegistry(address _newRegistry) external onlyRegistryOwner {
+        registry = Registry(_newRegistry);
+        emit RegistrySet(_newRegistry);
     }
 
-    function setRegistry(address _newRegistry) external {
-        require(msg.sender == REGISTRY.owner(), "Only registry owner can set");
-        REGISTRY = Registry(_newRegistry);
+    /// @notice _maxPriceDeviation becomes a % when divided by 10^18 (e.g. 10^17 becomes 10%)
+    function setMaxPriceDeviation(uint256 _maxPriceDeviation) external onlyRegistryOwner {
+        maxPriceDeviation = _maxPriceDeviation;
+        emit MaxPriceDeviationSet(_maxPriceDeviation);
     }
 
-    /// @notice _maxSpread becomes a percentage when divided by 10^18
-    /// @notice (e.g. 10^17 becomes 10%)
-    function setMaxSpread(uint _maxSpread) external {
-        require(msg.sender == REGISTRY.owner(), "Only registry owner can set");
-        MAX_SPREAD = _maxSpread;
+    /// @notice _maxSpread becomes a % when divided by 10^18 (e.g. 10^17 becomes 10%)
+    function setMaxSpread(uint256 _maxSpread) external onlyRegistryOwner {
+        maxSpread = _maxSpread;
+        emit MaxSpreadSet(_maxSpread);
     }
 
-    // PUBLIC VIEW METHODS
+    // PUBLIC/EXTERNAL VIEW FUNCTIONS
```

Comment:
Not really, just not totally sold on ordering functions strictly according to that rubric. This separates things like `getPrice public` and `getPrices external` across the file when it seems like they belong together and in that order. Also the style guide ordering and alphabetical ordering leads to functions being used before they are defined in a file, which is a bit backwards.
Will just go with the solidity style guide for now though since it's what people use and I don't want to litigate it any more.